src/main/java/edu/iis/powp/adapter/PlotterAdapter.java

```diff
@@ -4,39 +4,36 @@
 import edu.kis.powp.drawer.panel.DrawPanelController;
 import edu.kis.powp.drawer.shape.ILine;
 import edu.kis.powp.drawer.shape.LineFactory;
+import edu.iis.powp.appext.*;
 
 
-/**
- * Plotter adapter to drawer with several bugs. 
- */
-public class MyAdapter extends DrawPanelController implements IPlotter
+public class PlotterAdapter extends DrawPanelController implements IPlotter
 { 
 	private int startX = 0, startY = 0;
 	
-    public MyAdapter() {
+    public PlotterAdapter() {
 		super();
 	}
     
 	@Override
-    public void setPosition(int x, int y)
-    {
+    public void setPosition(int x, int y) {
         this.startX = x;
         this.startY = y;
     }
 
     @Override
-    public void drawTo(int x, int y)
-    {
+    public void drawTo(int x, int y) {
         ILine line = LineFactory.getBasicLine();
     	line.setStartCoordinates(this.startX, this.startY);
         line.setEndCoordinates(x, y);
+        this.startX = x;
+        this.startY = y;
 
-		drawLine(line);
+        ApplicationWithDrawer.getDrawPanelController().drawLine(line);
```

Comment:
1. Constructor injection should be used (using a parameter). This is unnecessary direct coupling with ApplicationWithDrawer class.
2. Inheritance is not used (no super drawLine calls). Should be removed!